pkgs/applications/kde/cantor.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, mkDerivation
+
+, cmake
+, extra-cmake-modules
+, makeWrapper
+, shared-mime-info
+
+, qtbase
+, qtsvg
+, qttools
+, qtwebengine
+, qtxmlpatterns
+
+, poppler
+
+, karchive
+, kcompletion
+, kconfig
+, kcoreaddons
+, kcrash
+, kdoctools
+, ki18n
+, kiconthemes
+, kio
+, knewstuff
+, kparts
+, kpty
+, ktexteditor
+, ktextwidgets
+, kxmlgui
+, syntax-highlighting
+
+, libspectre
+
+, backends ? [] # array of packages to provide for backends
+  # optional
+  # analitza,
+  # julia_17-bin,
+  # libqalculate,
+  # luajit,
+  # python3,
+  # R,
+  # sage,
```

Comment:
I was unsure if there was a better way to provide the backends since I think it links them in the output hence causing total output to grow

Reply:
Do we link against them or just need to add them to PATH?

Reply:
I do not know how to make it so that they are not linked. I inspected the strings in each of the backend binaries to see if they are linked

- `strings ./result/bin/.cantor_juliaserver-wrapped | rg /nix` contains two references to /nix/store/****-julia-1.8.4/lib
- `strings ./result/bin/.cantor_pythonserver-wrapped | rg /nix` contains a reference to /nix/store/****-python3-3.10.9/lib
- `strings ./result/bin/.cantor_rserver-wrapped | rg /nix | rg R-4.2.2` contains references to /nix/store/****-R-4.2.2/lib

(Note that I am looking at the `.***-wrapped` files due to the wraphook)

Also, for lua, libqalculate and octave,  there are no binaries in /bin.

Reply:
Currently cantor startup screen shows the following
![Screenshot from 2023-01-08 15-14-03](https://user-images.githubusercontent.com/53281855/211214496-2f97e2a2-afd1-4e76-a131-950847a536d1.png)

If I comment all of the backends out in the `buildInputs` list, I have the following
![Screenshot from 2023-01-08 15-15-23](https://user-images.githubusercontent.com/53281855/211214547-0ee6383a-d63f-426b-9c82-424b9b1f0134.png)